### PR TITLE
Clarifying polymorphic types mapping feature

### PR DESCRIPTION
**Pull request by glmars**

### CODE REVIEW

**Comment by bfil** [comment]
Thanks for this! Any chance you could add a test case as well?

**Comment by glmars** [comment]
A good idea! I'll try soon.

File: automapper/src/test/scala/io/bfil/automapper/AutomapperSpec.scala [hunk 1]
```diff
@@ -123,6 +123,52 @@ class AutomapperSpec extends WordSpec with Matchers with TestData {
 
   }
 
+  "automap polymorphic type" should {
+
+
```
Comment (bfil) [inline automapper/src/test/scala/io/bfil/automapper/AutomapperSpec.scala]: Please remove the extra empty line

File: automapper/src/test/scala/io/bfil/automapper/AutomapperSpec.scala [hunk 2]
```diff
@@ -123,6 +123,52 @@ class AutomapperSpec extends WordSpec with Matchers with TestData {
 
   }
 
+  "automap polymorphic type" should {
```
Comment (bfil) [inline automapper/src/test/scala/io/bfil/automapper/AutomapperSpec.scala]: Change to "automap polymorphic types" please

File: automapper/src/test/scala/io/bfil/automapper/AutomapperSpec.scala [hunk 3]
```diff
@@ -123,6 +123,52 @@ class AutomapperSpec extends WordSpec with Matchers with TestData {
...
+
+    "map a polymorphic type field really" in {
+
+      implicit val conversion = mapPolymorphicTrait _
+
+      automap(sourcePolymorphicB).to[TargetPolymorphicClass] === targetPolymorphicB
+
+    }
+
+    "throw exception on unknown class" in {
```
Comment (bfil) [inline automapper/src/test/scala/io/bfil/automapper/AutomapperSpec.scala]: Please rename to "throw an exception for an unmapped polymorphic type"

File: automapper/src/test/scala/io/bfil/automapper/AutomapperSpec.scala [hunk 4]
```diff
@@ -123,6 +123,52 @@ class AutomapperSpec extends WordSpec with Matchers with TestData {
...
+
+      automap(sourcePolymorphicA).to[TargetPolymorphicClass] === targetPolymorphicA
+
+    }
+
+    "map a polymorphic type field really" in {
+
+      implicit val conversion = mapPolymorphicTrait _
+
+      automap(sourcePolymorphicB).to[TargetPolymorphicClass] === targetPolymorphicB
```
Comment (bfil) [inline automapper/src/test/scala/io/bfil/automapper/AutomapperSpec.scala]: This line could be put in the previous test case

File: automapper/src/test/scala/io/bfil/automapper/AutomapperSpec.scala [hunk 5]
```diff
@@ -123,6 +123,52 @@ class AutomapperSpec extends WordSpec with Matchers with TestData {
...
+
+        implicit val conversion = mapPolymorphicTrait _
+
+        automap(sourcePolymorphicC).to[TargetPolymorphicClass]
+
+      }
+
+    }
+
+    "not compile without implicit conversion in scope" in {
```
Comment (bfil) [inline automapper/src/test/scala/io/bfil/automapper/AutomapperSpec.scala]: "not compile without an implicit conversion in scope"

**Review by bfil — COMMENTED** [review]
I've added a few comments, it's mostly about wording in the tests, nothing time-consuming. Thanks again!

File: automapper/src/test/scala/io/bfil/automapper/AutomapperSpec.scala (hunk 1 above)
Comment (glmars) [reply automapper/src/test/scala/io/bfil/automapper/AutomapperSpec.scala]: Oops, I'm sorry.

**Comment by glmars** [comment]
Thank you for your comments, I fixed them in an additional commit.

**Pull request merged**